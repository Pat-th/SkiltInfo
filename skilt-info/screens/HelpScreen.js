import React from 'react';
import {ScrollView, View} from 'react-native';
import Expandables from "../components/Expandables";

const HelpScreen = () => {
    const name = [
        {
            title: 'Bruk',
            content: 'Ta bilde av et veiskilt fra kamerasiden og vent på at bildet skal gjenkjennes og kartet skal åpne seg. Velg så skiltet ditt fra markørene på kartet. Velger du feil skilt kan du trykke på tilbakeknappen og velge et annet.',
        },
        {
            title: 'Skiltet mitt er ikke på kartet',
            content: 'Vennligst prøv å ta et nytt bilde i tilfelle vi har gjort en feil i gjenkjenning av bildet',
        },
        {
            title: 'Sikkerhet',
            content: 'Applikasjonen lagrer ingen data fra brukerne. Bilder som tas i applikasjonen og posisjonen til brukeren brukes kun til å finne riktig skilt',
        },
        {
            title: 'Info og Opphavsrett',
            content: 'Applikasjonen er utviklet av Patrick Thorkildsen og Quan Tran i samarbeid med Triona AS som del av bacheloroppgaven på dataingeniørstudiet ved NTNU i 2020',
        }
    ];

    return (
        <View>
            <ScrollView>
                <Expandables sections={name}/>
            </ScrollView>
        </View>
    );
};


export default HelpScreen;

