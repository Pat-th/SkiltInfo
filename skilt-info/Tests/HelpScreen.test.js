import React from "react";
import renderer from "react-test-renderer";

import HelpScreen from "../screens/HelpScreen";

describe("<HelpScreen />", () => {
    it("Has 1 child", () => {
        const tree = renderer.create(<HelpScreen />).toJSON();
        expect(tree.children.length).toBe(2);
    });
    it('Renders correctly', () => {
        const tree = renderer.create(<HelpScreen />).toJSON();
        expect(tree).toMatchSnapshot();
      });
   });