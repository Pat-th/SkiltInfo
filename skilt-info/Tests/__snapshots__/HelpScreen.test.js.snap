// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelpScreen /> Renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "height": 80,
        "justifyContent": "center",
        "paddingTop": 36,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 18,
        }
      }
    >
      Hjelp
    </Text>
  </View>
  <RCTScrollView>
    <View>
      <View>
        <RCTScrollView>
          <View>
            <View>
              <View
                style={Object {}}
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "rgb(255,255,255)",
                        "borderColor": "rgb(0,0,0)",
                        "borderRadius": 0,
                        "borderWidth": 3,
                        "color": "rgba(0,0,0,1)",
                        "justifyContent": "center",
                        "marginTop": 6,
                        "padding": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "rgba(0,0,0,1)",
                          "fontSize": 24,
                          "fontWeight": "500",
                          "textAlign": "left",
                        }
                      }
                    >
                      Hvordan bruke appen
                    </Text>
                    <Image
                      source={1}
                      style={
                        Object {
                          "height": 30,
                          "position": "absolute",
                          "right": 10,
                          "width": 30,
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "height": 0,
                      "overflow": "hidden",
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    style={Object {}}
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "rgb(255,255,255)",
                          "borderColor": "rgb(0,0,0)",
                          "borderRadius": 0,
                          "borderWidth": 3,
                          "color": "rgba(0,0,0,1)",
                          "marginTop": 6,
                          "padding": 5,
                          "textAlign": "left",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "rgba(0,0,0,1)",
                            "padding": 5,
                            "textAlign": "left",
                          }
                        }
                      >
                        Trykk på kameraknappen på bunnen av appen. Trykk på sirkelen rett over kameraknappen. Du vil så få et valg over
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={Object {}}
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "rgb(255,255,255)",
                        "borderColor": "rgb(0,0,0)",
                        "borderRadius": 0,
                        "borderWidth": 3,
                        "color": "rgba(0,0,0,1)",
                        "justifyContent": "center",
                        "marginTop": 6,
                        "padding": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "rgba(0,0,0,1)",
                          "fontSize": 24,
                          "fontWeight": "500",
                          "textAlign": "left",
                        }
                      }
                    >
                      Vanlige feil
                    </Text>
                    <Image
                      source={1}
                      style={
                        Object {
                          "height": 30,
                          "position": "absolute",
                          "right": 10,
                          "width": 30,
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "height": 0,
                      "overflow": "hidden",
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    style={Object {}}
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "rgb(255,255,255)",
                          "borderColor": "rgb(0,0,0)",
                          "borderRadius": 0,
                          "borderWidth": 3,
                          "color": "rgba(0,0,0,1)",
                          "marginTop": 6,
                          "padding": 5,
                          "textAlign": "left",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "rgba(0,0,0,1)",
                            "padding": 5,
                            "textAlign": "left",
                          }
                        }
                      >
                        Unngå å stå i veikryss eller under/på bruer.
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={Object {}}
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "rgb(255,255,255)",
                        "borderColor": "rgb(0,0,0)",
                        "borderRadius": 0,
                        "borderWidth": 3,
                        "color": "rgba(0,0,0,1)",
                        "justifyContent": "center",
                        "marginTop": 6,
                        "padding": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "rgba(0,0,0,1)",
                          "fontSize": 24,
                          "fontWeight": "500",
                          "textAlign": "left",
                        }
                      }
                    >
                      Opphavsrett
                    </Text>
                    <Image
                      source={1}
                      style={
                        Object {
                          "height": 30,
                          "position": "absolute",
                          "right": 10,
                          "width": 30,
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "height": 0,
                      "overflow": "hidden",
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    style={Object {}}
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "rgb(255,255,255)",
                          "borderColor": "rgb(0,0,0)",
                          "borderRadius": 0,
                          "borderWidth": 3,
                          "color": "rgba(0,0,0,1)",
                          "marginTop": 6,
                          "padding": 5,
                          "textAlign": "left",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "rgba(0,0,0,1)",
                            "padding": 5,
                            "textAlign": "left",
                          }
                        }
                      >
                        test
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={Object {}}
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "rgb(255,255,255)",
                        "borderColor": "rgb(0,0,0)",
                        "borderRadius": 0,
                        "borderWidth": 3,
                        "color": "rgba(0,0,0,1)",
                        "justifyContent": "center",
                        "marginTop": 6,
                        "padding": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "rgba(0,0,0,1)",
                          "fontSize": 24,
                          "fontWeight": "500",
                          "textAlign": "left",
                        }
                      }
                    >
                      Heisann Patrick
                    </Text>
                    <Image
                      source={1}
                      style={
                        Object {
                          "height": 30,
                          "position": "absolute",
                          "right": 10,
                          "width": 30,
                        }
                      }
                    />
                  </View>
                </View>
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "height": 0,
                      "overflow": "hidden",
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    style={Object {}}
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "rgb(255,255,255)",
                          "borderColor": "rgb(0,0,0)",
                          "borderRadius": 0,
                          "borderWidth": 3,
                          "color": "rgba(0,0,0,1)",
                          "marginTop": 6,
                          "padding": 5,
                          "textAlign": "left",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "rgba(0,0,0,1)",
                            "padding": 5,
                            "textAlign": "left",
                          }
                        }
                      >
                        heihei
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
